tests/loggers/test_cometml_logger.py

```diff
@@ -1,15 +1,122 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
-
 import os
 import zipfile
+from collections import defaultdict
 from json import JSONDecoder
 from pathlib import Path
 
 import pytest
+from torch.utils.data import DataLoader
+
+from composer.trainer import Trainer
+from tests.common import RandomClassificationDataset, SimpleModel
+
+
+def test_comet_ml_logging_train_loop(monkeypatch, tmp_path):
+    pytest.importorskip('comet_ml', reason='comet_ml is optional')
+    import comet_ml
+
+    monkeypatch.setattr(comet_ml, 'Experiment', comet_ml.OfflineExperiment)
+    from composer.loggers import CometMLLogger
+
+    # Set offline directory.
+    offline_directory = str(tmp_path / Path('.my_cometml_runs'))
+    os.environ['COMET_OFFLINE_DIRECTORY'] = offline_directory
+
+    comet_logger = CometMLLogger()
+
+    trainer = Trainer(
+        model=SimpleModel(),
+        train_dataloader=DataLoader(RandomClassificationDataset()),
+        train_subset_num_batches=2,
+        max_duration='2ep',
+        loggers=comet_logger,
+    )
+    trainer.fit()
+
+    del trainer
+
+    assert comet_logger.experiment is not None
+
+    # Open, decompress, decode, and extract offline dump of metrics.
+    comet_exp_dump_filepath = str(Path(offline_directory) / Path(comet_logger.experiment.id).with_suffix('.zip'))
+    zf = zipfile.ZipFile(comet_exp_dump_filepath)
+    comet_logs_path = zf.extract('messages.json', path=offline_directory)
+    jd = JSONDecoder()
+    msg_type_to_msgs = defaultdict(list)
+
+    with open(comet_logs_path) as f:
+        for line in f.readlines():
+            parsed_line = jd.decode(line)
+            msg_type_to_msgs[parsed_line['type']].append(parsed_line['payload'])
+
+    # Check that init set the run name
+    assert comet_logger.name is not None
+    assert comet_logger.experiment.name is not None
```

Comment:
The name won't be None if it's not set. It will actually be a randomly chosen name, so the best way to check if it is set is to check if it's value is correct.
```suggestion
    assert comet_logger.experiment.name == trainer.state.run_name
```
You'd have to remove the `del trainer` line here also.

Reply:
ah good point. I have to `del trainer` so that `experiment.end()` is called, but I'll just save the correct name before that.